api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -1980,4 +1980,43 @@ public boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType)
 		}
 		return mappedClasses;
 	}
+
+	@Override
+	public Concept getConceptByUuidOrMapping(String uuid, String code, String sourceName) throws APIException {
+		return getConceptByUuidOrMapping(uuid, code, sourceName, true);
+	}
+
+	@Override
+	public Concept getConceptByUuidOrMapping(String uuid, String code, String sourceName, boolean includeRetired)
+	        throws APIException {
+		List<Concept> concepts = Context.getConceptService().getConceptsByUuidOrMapping(uuid,code, sourceName, includeRetired);
+		
+		if (concepts.isEmpty()) {
+			return null;
+		}
+		else if (concepts.size() > 1 && !concepts.get(1).getRetired()) {
+			throw new APIException("Concept.error.multiple.non.retired", new Object[] { code, sourceName });
+		} else {
+			return concepts.get(0);
+					}	
+	}
+
+	@Override
+	public List<Concept> getConceptsByUuidOrMapping(String uuid, String code, String sourceName) throws APIException {
+		return getConceptsByUuidOrMapping(uuid, code, sourceName, true);
+	}
+
+	@Override
+	public List<Concept> getConceptsByUuidOrMapping(String uuid, String code, String sourceName, boolean includeRetired)
```

Comment:
@reagan-meant  getConceptByUuid and getConceptByMapping both methods does not return a list of Concept. Is there any particular reason to return a list here ?

Reply:
the idea for returning a list is to find out if there are retired concepts thus in all these methods they call even the retired concepts

Reply:
You are using getConceptByUuid and getConceptByMapping to get the Concepts. But none of them returns a list.

Reply:
@dilanthas i get your point now...however, what are your suggestions on how to approach this then because in my method as per the ticket I need to return values with different types..one is list and the other is not

Reply:
@dilanthas seems like the easier way would be to create a new method for getConceptByUuuid with a return type of List<Concept>

Reply:
@reagan-meant Sorry i didn't get you.  What i meant was in line 2015 you are calling getConceptByUuid and it returns a Concept. not a List<Concept>.